stdlib/Random/src/RNGs.jl

```diff
@@ -290,20 +290,51 @@ function make_seed()
     end
 end
 
+"""
+    make_seed(n::Integer) -> Vector{UInt32}
+
+Transform `n` into a bit pattern encoded as a `Vector{UInt32}`, suitable for
+RNG seeding routines.
+
+`make_seed` is "injective" : if `n != m`, then `make_seed(n) != `make_seed(m)`.
+Moreover, if `n == m`, then `make_seed(n) == make_seed(m)`.
+
+This is an internal function, subject to change.
+"""
 function make_seed(n::Integer)
-    n < 0 && throw(DomainError(n, "`n` must be non-negative."))
+    neg = signbit(n)
+    n = abs(n) # n can still be negative, e.g. n == typemin(Int)
+    if n < 0
+        # we assume that `unsigned` can be called on integers `n` for which `abs(n)` is
+        # negative; `unsigned` is necessary for `n & 0xffffffff` below, which would
+        # otherwise propagate the sign bit of `n` for types smaller than UInt32
+        n = unsigned(n)
```

Comment:
Does this introduce a type instability?

Reply:
Yes, but this should be handled fine by union-splitting.